### ISSUE TITLE
Feature Request: Does `evaluate()` intentionally calculate a "mean of means" for global metrics?

### ISSUE THREAD

**Issue by cespeleta** [Nixtla/utilsforecast#252]
### Description

I am opening this issue to ask about the expected behavior of the `evaluate` function when attempting to calculate a global forecast accuracy metric (e.g., global RMSE across all series).

When evaluating the forecast accuracy per series (determined by the `unique_id` column), the RMSE is correct and matches the output from other libraries like `scikit-learn`. However, when we want to obtain the total/global metric using `agg_fn="mean"`, `evaluate` seems to calculate the average of the per-series RMSEs (a "mean of means") rather than the true global RMSE over all data points.

### Reproducible Example
Here is a code snippet demonstrating the difference between the global calculation in `utilsforecast` and `scikit-learn`:

```python
import numpy as np
from sklearn.metrics import root_mean_squared_error
from utilsforecast.data import generate_series
from utilsforecast.evaluation import evaluate
from utilsforecast.losses import rmse

# Generate mock data
series = generate_series(10, with_trend=True, static_as_categorical=False)
valid = series.groupby("unique_id").tail(7).copy()
train = series.drop(valid.index)

rng = np.random.RandomState(0)
valid["seas_naive"] = train.groupby("unique_id")["y"].tail(7).values

# 1. Global RMSE: Differing results
print("--- Global RMSE ---")
global_nixtla = evaluate(valid, metrics=[rmse], agg_fn="mean")
print(f"utilsforecast:\n{global_nixtla.to_string()}\n")
# utilsforecast:
# metric  seas_naive
# rmse    2.950459

global_sklearn = root_mean_squared_error(valid["y"], valid["seas_naive"])
print(f"scikit-learn: {global_sklearn}\n")
# scikit-learn: 3.4624764649542468

# 2. Metric by unique_id: Matches perfectly
print("--- RMSE per unique_id ---")
per_series_nixtla = evaluate(valid, metrics=[rmse])
print(f"utilsforecast:\n{per_series_nixtla.to_string()}\n")
# utilsforecast:
#    unique_id   metric  seas_naive
# 0         0      rmse    3.346506
# 1         1      rmse    0.888731
# ...

per_series_sklearn = valid.groupby("unique_id").apply(
    lambda x: root_mean_squared_error(x["y"], x["seas_naive"]), include_groups=False
)
print(f"scikit-learn:\n{per_series_sklearn.to_string()}\n")
# scikit-learn:
#     unique_id
# 0    3.346506
# 1    0.888731
# ...

# 3. Average of RMSEs
print("--- Average of per-series RMSEs ---")
print(f"utilsforecast mean: {per_series_nixtla['seas_naive'].mean()}")
print(f"scikit-learn mean:  {per_series_sklearn.mean()}")
# --- Average of per-series RMSEs ---
# utilsforecast mean: 2.9504585011788462
# scikit-learn mean:  2.9504585011788462
```
### Expected vs. Actual Behavior
* **Actual Behavior**: `evaluate(..., agg_fn="mean")` groups by `id_col`, calculates the metric for each series, and then averages those results.
* **Question**: Is this the intended behavior for global evaluation within the Nixtla ecosystem?

### Proposed Solution

If calculating the true global error (treating all rows as a single dataset regardless of `unique_id`) is a desired feature, a potential solution would be to allow the `id_col` argument to accept `None`.

If `id_col=None` is passed to `evaluate`, it could bypass the groupby operation and directly calculate the global error across the entire dataframe.

Let me know your thoughts on this!

**Comment by cespeleta** [comment]
Any thoughts?

**Comment by jmoralez** [comment]
Hey! Yes, this is the expected behavior, the `agg_fn` is applied to the metrics by serie.

If you want to treat all observations as a single serie you could set a dummy value to `id_col`, e.g. `global_nixtla = evaluate(valid.assign(unique_id=0), metrics=[rmse], agg_fn="mean")`, but that'll do more work than just computing the metric directly on the two columns. Do you want to produce these global results for several metrics?

I'm not sure how common it is to treat all series equally, that's why the default of `agg_fn` is `None`, to let the user decide if/how they want to aggregate the individual series metrics, because they usually have different scales.

**Comment by cespeleta** [comment]
Hi @jmoralez, thanks for the example - it works perfectly.
